demo/app/routes/muc/muc.component.html

```diff
@@ -0,0 +1,84 @@
+                    <input [(ngModel)]="topic" class="form-control" id="room-topic" name="room-topic"
+                           placeholder="Enter a new room subject" type="text"/>
+                    <button type="button" (click)="changeRoomTopic()">Change</button>
+                </div>
+                <div class="form-group">
+                    <label for="change-nick">Change nick:</label>
+                    <input [(ngModel)]="nick" class="form-control" id="change-nick" name="change-nick"
+                           placeholder="Enter a new nick" type="text"/>
+                    <button type="button" (click)="changeNick()">Change</button>
+                </div>
+                <div class="form-group">
+                    <label for="invite-user">Invite user:</label>
+                    <input [(ngModel)]="inviteJid" class="form-control" id="invite-user" name="invite-user"
+                           placeholder="Enter JID of a user to invite" type="text"/>
+                    <button type="button" (click)="inviteUser()">Invite</button>
+                </div>
+                <div class="form-group">
+                    <label for="member">Grant or revoke membership</label>
+                    <input [(ngModel)]="memberJid" class="form-control" id="member" name="member" type="text" placeholder="Enter JID of an occupant to grant/revoke membership" />
+                    <button type="button" (click)="revokeMembership()">Revoke</button>
+                    <button type="button" (click)="grantMembership()">Grant</button>
+                </div>
+                <div class="form-group">
+                    <label for="moderator">Grant or revoke moderator status</label>
+                    <input [(ngModel)]="moderatorJid" class="form-control" id="moderator" name="moderator" type="text" placeholder="Enter nick of an occupant to grant or revoke moderator status"/>
+                    <button type="button" (click)="revokeModeratorStatus()">Revoke</button>
+                    <button type="button" (click)="grantModeratorStatus()">Grant</button>
+                </div>
+            </form>
+        </ng-container>
+        <ng-container *ngIf="selectedRoom$ | async as room">
+            <h2>Chat:</h2>
+            <ngx-chat-message-list [recipient]="room"></ngx-chat-message-list>
+            <ngx-chat-message-input [recipient]="room"></ngx-chat-message-input>
+        </ng-container>
+    </main>
+    <aside>
+        <h4>Occupants:</h4>
+        <div class="list">
+            <div class="list-item" *ngFor="let occupant of occupants">
+                <div>
+                    <p>{{occupant.nick}} - <small>{{occupant.affiliation}}, {{occupant.role}}</small></p>
+                    <small>{{occupant.jid}}</small>
+                </div>
+                <div>
+                    <button (click)="kick(occupant)">Kick</button>
+                    <button (click)="ban(occupant)">Ban</button>
```

Comment:
Banning users won't work with user's occupant JIDs (see: https://xmpp.org/extensions/xep-0045.html#ban, "The ban MUST be performed based on the occupant's bare JID."). You'll need to fetch their bare JID. For users with affiliation/role != none this is simple but I have no idea how to do it for users with affiliation/role of none.

Reply:
This should work now